src/librustdoc/clean/mod.rs

```diff
@@ -1080,7 +1081,9 @@ impl Clean<Item> for ty::AssocItem {
                             constness: hir::Constness::NotConst,
                             asyncness: hir::IsAsync::NotAsync,
                         },
-                    })
+                        def_id: self.def_id,
+                    };
+                    TyMethodItem(function)
```

Comment:
nit: I think you can revert adding the local variables; it doesn't really matter though.